web/src/pages/api/public/ingestion.ts

```diff
@@ -549,47 +549,3 @@ export const aggregateBatchResult = (
 
   return { successes, errors: returnedErrors };
 };
-
-/**
- * Parses the response from the ingestion batch API event processor and throws an error of `BaserError` if the response is not as expected.
- *
- * @param errors - Array of errors from `handleBatch()`
- * @param results - Array of results from `handleBatch()`
- * @param object - Zod object to parse the result, if not provided, the result is returned as is without parsing
- * @returns - Parsed result
- * @throws - Throws an error of type `BaseError` if there are errors in the arguments
- */
-export const parseSingleTypedIngestionApiResponse = <T extends z.ZodTypeAny>(
-  errors: Array<{ id: string; error: unknown }>,
-  results: Array<{ id: string; result: unknown }>,
-  object?: T,
-): T extends z.ZodTypeAny ? z.infer<T> : unknown => {
-  const unknownErrors = errors.map((error) => error.error);
-  const badRequestErrors = getBadRequestError(unknownErrors);
-  if (badRequestErrors.length > 0) {
-    throw new InvalidRequestError(badRequestErrors[0].message);
-  }
-  const langfuseNotFoundError = getLangfuseNotFoundError(unknownErrors);
-  if (langfuseNotFoundError.length > 0) {
-    throw langfuseNotFoundError[0];
-  }
-  if (errors.length > 0) {
-    throw new InternalServerError("Internal Server Error");
-  }
-
-  if (results.length === 0) {
-    throw new InternalServerError("No results returned");
-  }
-
-  if (object === undefined) {
-    return results[0].result as T extends z.ZodTypeAny ? z.infer<T> : unknown;
-  }
-
-  const parsedObj = object.safeParse(results[0].result);
```

Comment:
i think all the errors here above return different http status codes. Anything against using parseSingleTypedIngestionApiResponse?

Reply:
@maxdeichmann We should already cover the status codes via https://github.com/langfuse/langfuse/blob/main/web/src/pages/api/public/ingestion.ts#L491 and the error.status property. IMO the parseSingleTypedIngestionApiResponse is fully covered by aggregateBatchResults.